### PR TITLE
fix: use latest ruby and newer jekyll versions

### PR DESCRIPTION
**Pull request by jmeridth**
The resulting `https://localhost:4000` redirects to `https://localhost:4000/feed.json` and it includes the whistleblowing content.  Still looking into that.

Screenshot 2024-10-12 at 1 37 37 PM

My changes listed below in each section under `My Changes`

### Please confirm this pull request meets the following requirements:

- [x] I followed the contributing guidelines: <https://github.com/github/explore/blob/main/CONTRIBUTING.md>.
- [x] I am not the sole author or employee of a company who created either the topic I am modifying/adding or the collection entry I am modifying/adding.

### Which change are you proposing?

  - [x] Suggesting edits to an existing topic or collection
  - [ ] Curating a new topic or collection
  - [x] Something that does not neatly fit into the binary options above

---

### Editing an existing topic or collection

I'm suggesting these edits to an existing topic or collection:
- [ ] Image (and my file is `*.png`, square, dimensions 288x288, size <= 75 kB)
- [x] Content (and my changes are in `index.md`)

#### My Changes

- [x] fix liquid parsing issue of handlebars/index.md (backticks to quotes)
  - [Liquid Warning example](https://github.com/github/explore/actions/runs/11299945094/job/31431925940#step:6:82)

### Curating a new topic or collection

- [ ] I've formatted my changes as a new folder directory, named for the topic or collection as it appears in the URL on GitHub (e.g. `https://github.com/topics/[NAME]` or `https://github.com/collections/[NAME]`)
- [ ] My folder contains a `*.png` image (if applicable) and `index.md`
- [ ] All required fields in my `index.md` conform to the Style Guide and API docs: <https://github.com/github/explore/tree/main/docs>

> Please replace this line with an explanation of why you think this topic or collection should be curated.

### Something that does not neatly fit into the binary options above

- [ ] My suggested edits are not about an existing topic or collection, or at least not a single one
- [x] My suggested edits are not about curating a new topic or collection, or at least not a single one
- [x] My suggested edits conform to the Style Guide and API docs: https://github.com/github/explore/tree/main/docs

#### My Changes

- [x] upgrade to ruby 3.3.5 from 3.1.2, update .ruby-version ([3.1 is end of life in 5 months](https://endoflife.date/ruby))
- [x] use latest github-pages gem which uses Jekyll 3.10.0 (from 6/2024)
  - previous was using jekyll 2.4.0 (was from 10 years ago)
- [x] setup .devcontainer for local development
- [x] add Gemfile.lock (remove from .gitignore) so we can update dependabot config to include gems/bundler (weekly)
- [x] fix markdown formatting in CONTRIBUTING.md

---

**Please note: we will close your PR without comment if you do not check the boxes above and provide ALL requested information.**

### CODE REVIEW

File: topics/handlebars/index.md [hunk 1]
```diff
@@ -10,4 +10,4 @@ related: mustache-js, ejs-templates, pug, nunjucks, jinja2, liquid, javascript,
 released: February 15, 2011
 github_url: https://github.com/handlebars-lang
 ---
-**Handlebars** is a simple templating language. It uses a template and an input object to generate HTML or other text formats. Handlebars templates look like regular text with embedded Handlebars expressions. Handlebars is largely compatible with Mustache templates. In most cases it is possible to swap out Mustache with Handlebars and continue using your current templates. A handlebars expression is a `{{`, some contents, followed by a `}}`. When the template is executed, these expressions are replaced with values from an input object.
+**Handlebars** is a simple templating language. It uses a template and an input object to generate HTML or other text formats. Handlebars templates look like regular text with embedded Handlebars expressions. Handlebars is largely compatible with Mustache templates. In most cases it is possible to swap out Mustache with Handlebars and continue using your current templates. A handlebars expression is a "{{", some contents, followed by a "}}". When the template is executed, these expressions are replaced with values from an input object.
```
Comment (jmeridth) [inline topics/handlebars/index.md:13]: changing from backticks to quotes fixed [liquid warning](https://github.com/github/explore/actions/runs/11299945094/job/31431925940#step:6:82)

**Comment by jmeridth** [comment]
This might be too much for one PR. 🤷

**Comment by ZARAH731** [comment]
[https://www.google.com/finance/markets/indexes/authorize?copy_id=3cc6375c-cbb0-44be-b697-747d4a034f88](url)

**Comment by ZARAH731** [comment]
> [https://www.google.com/finance/markets/indexes/authorize?copy_id=3cc6375c-cbb0-44be-b697-747d4a034f88](url)

File: topics/handlebars/index.md (hunk 1 above)
Comment (ZARAH731) [reply topics/handlebars/index.md:13]: @@ -10,4 +10,4 @@ related: mustache-js, ejs-templates, pug, nunjucks, jinja2, liquid, javascript,
released: February 15, 2011
github_url: https://github.com/handlebars-lang
---
**Handlebars** is a simple templating language. It uses a template and an input object to generate HTML or other text formats. Handlebars templates look like regular text with embedded Handlebars expressions. Handlebars is largely compatible with Mustache templates. In most cases it is possible to swap out Mustache with Handlebars and continue using your current templates. A handlebars expression is a `{{`, some contents, followed by a `}}`. When the template is executed, these expressions are replaced with values from an input object.
**Handlebars** is a simple templating language. It uses a template and an input object to generate HTML or other text formats. Handlebars templates look like regular text with embedded Handlebars expressions. Handlebars is largely compatible with Mustache templates. In most cases it is possible to swap out Mustache with Handlebars and continue using your current templates. A handlebars expression is a "{{", some contents, followed by a "}}". When the template is executed, these expressions are replaced with values from an input object.

**Review by tomthorogood — APPROVED** [review]
✨

**Comment by evilaliv3** [comment]
@jmeridth @tomthorogood : I see that this patch has been included but still the topic [whistleblowing](https://github.com/topics/whistleblowing) appears not working.

Have you identified the reason ?

**Comment by evilaliv3** [comment]
@jmeridth @tomthorogood ?

**Comment by jmeridth** [comment]
> @jmeridth @tomthorogood ?

@evilaliv3 We are still looking into this.  Your topic isn't the only one affected.  Will update once we have something.  Thank you for your patience.

**Comment by evilaliv3** [comment]
Thank you @jmeridth !

**Comment by evilaliv3** [comment]
@jmeridth :  This is a kind reminder to look into the problem causing some topics to not be loaded correctly.

Do you suggest we open a dedicated ticket? I might open it but i let it to you as you probably already know how to specify it.

Thank you for your work!

**Comment by evilaliv3** [comment]
@jmeridth @tomthorogood , is there any update on the issue about the topics not showing up?

Maybe there are checks on "bad" words? whistleblowing some times is wrongly included in such lists

**Pull request merged**